scripts/manageLAND.js

```diff
@@ -67,45 +114,130 @@ const manageMana = {
+    gasPrice: options.gasPrice,
+    txDelay: options.txDelay,
+    retryFailedTxs: options.retryFailedTxs,
+    recipients
+  })
+}
+
+async function transferAllAction(options) {
+  checkContains(options, requiredTransferAllOptionNames)
+
+  await setupEth({
+    account: options.account,
+    password: options.password,
+    providerType: options.provider
+  })
+
+  const balance = await getBalance(options.from)
+  const { from, to } = options
+
+  if (!options.yes) {
+    const shouldTransfer = await cli.confirm(
+      `About to transfer ${balance} LANDs from ${from} to ${to} by ${eth.getAccount()}. Ok?`
+    )
+    if (!shouldTransfer) process.exit()
+  }
+
+  const landBatches = await getLANDBatches(
+    from,
+    balance.toNumber(),
+    options.landBatchSize
+  )
 
-          const landBatches = await getLANDBatches(
+  await transferManyLAND({
+    batchSize: options.batchSize,
+    gasPrice: options.gasPrice,
+    txDelay: options.txDelay,
+    retryFailedTxs: options.retryFailedTxs,
+    from,
+    to,
+    landBatches
+  })
+}
+
+async function transferLANDs(args) {
+  const { recipients, batchSize, gasPrice, shouldRetry, txDelay } = args
+
+  const account = eth.getAccount()
+  const LANDRegistryContract = eth.getContract('LANDRegistry')
+
+  log.info('Transfering LANDs')
+
+  const txsToRetry = await executeTransactions(
+    recipients,
+    recipientsBatch =>
+      recipientsBatch.map(async recipient => {
+        const { from, to, id } = recipient
+        const [x, y] = id.split(',')
+        const encodedId = await LANDRegistryContract.encodeTokenId(x, y)
+        if (from.toLowerCase() !== account.toLowerCase()) {
+          const isApprovedForAll = await LANDRegistryContract.isApprovedForAll(
             from,
-            balance.toNumber(),
-            options.landBatchSize
+            account
           )
+          if (!isApprovedForAll) {
+            await checkAllowance({ id: encodedId, x, y })
+          }
+        }
 
-          await transferLANDs({
-            batchSize: options.batchSize,
-            gasPrice: options.gasPrice,
-            txDelay: options.txDelay,
-            retryFailedTxs: options.retryFailedTxs,
-            from,
-            to,
-            landBatches
-          })
-        })
-      )
+        console.log(`Sending (${x}, ${y}) from ${from} to ${to}\n`)
+
+        const hash = await LANDRegistryContract.transferFrom(
+          from,
+          to,
+          encodedId,
+          {
+            gasPrice: gasPrice,
+            from: account
+          }
+        )
+
+        return { hash, data: recipient }
+      }),
+    { batchSize, txDelay }
+  )
+
+  log.info(`Sent ${recipients.length - txsToRetry.length} transactions`)
+
+  if (txsToRetry.length > 0 && shouldRetry) {
+    const recipientsToRetry = txsToRetry.map(tx => tx.data)
+    log.info(`Retrying on ${recipientsToRetry.length} recipients`)
+    return transferLANDs(recipientsToRetry, ...args.slice(1))
```

Comment:
Sorry for the silly question, but shouln't `args` be an object? (without a `slice` method)

Reply:
Wow! good catch!